src/Filters.cpp

```diff
@@ -175,7 +175,7 @@ float Filters::update_voltage_supercap(float v) {
 }
 
 void Filters::correctAltitude(struct DataFrame &data){
-  if(data.NUM_SATS_GPS >= 6){
+  if( (data.NUM_SATS_GPS >= 6) && (data.GPS_LAST_NEW - data.TIME < 1000*60*5) && (fabs(data.ALTITUDE_GPS - data.ALTITUDE_BAROMETER) < 1000)) {
     data.ALTITUDE_OFFSET = data.ALTITUDE_OFFSET*(1-data.ALTITUDE_OFFSET_GAIN) + (data.ALTITUDE_GPS - data.ALTITUDE_BAROMETER)*data.ALTITUDE_OFFSET_GAIN;
```

Comment:
can we protect against unknown dumb errors by bounding the offset to, say, |offset|<1km, or whatever you think is practical from past flight experience? Just having that makes me way less worried about all sorts of GPS errors because their effect becomes well bounded.

Reply:
Well, the fact that `ALTITUDE_OFFSET` offset will not update if the abs difference is greater than 1km ensures this. When `ALTITUDE_OFFSET` is updated, it is slightly nudge towards the value of `data.ALTITUDE_GPS - data.ALTITUDE_BAROMETER`. And, it will only update if `fabs(data.ALTITUDE_GPS - data.ALTITUDE_BAROMETER) < 1000)`

Reply:
ohp, I was dumb and missed that.